### ISSUE TITLE
firebase-admin-node needs to declare a dependency on @types/google-cloud__storage

### ISSUE THREAD

**Issue by tikurahul** [firebase/firebase-admin-node#92]
### [REQUIRED] Step 2: Describe your environment

  * Operating System macOS 10.13 (High Seirra)
  * Firebase Admin Node 5.3.0
  * Firebase Product: Firebase Functions

### [REQUIRED] Step 3: Describe the problem

`firebase-admin` needs to declare a dependency on `@types/google-cloud__storage` for TypeScript projects.

#### Steps to reproduce:

* Define a Typescript project which depends on `firebase-functions`.
* Install the `firebase-admin` node package (peer dependency)
* `tsc` fails with this error.

```node_modules/firebase-admin/lib/index.d.ts(18,22): error TS7016: Could not find a declaration file for module '@google-cloud/storage'. '/Users/rahulrav/Workspaces/Opensource/smart-interviews-backend/node_modules/firebase-admin/node_modules/@google-cloud/storage/src/index.js' implicitly has an 'any' type.
  Try `npm install @types/@google-cloud/storage` if it exists or add a new declaration (.d.ts) file containing `declare module '@google-cloud/storage';`
```

**Comment by hiranya911** [comment]
We declared `@types/google-cloud__storage` as a dev dependency since it never gets referenced during actual runtime. We have been asking TypeScript developers to install it separately as needed. But perhaps we should make it a regular dependency so the TypeScript developers get a better experience out of the box.

**Comment by tikurahul** [comment]
That would be awesome. It's easy enough to install it. But consumers should not have to install types that are a dependency for firebase-admin-console as this list of types increase.

**Comment by hiranya911** [comment]
Does your code make any references to Cloud Storage types? Perhaps you can post your package.json and a code snippet?

**Comment by tikurahul** [comment]
```json
{
  "name": "...",
  "version": "...",
  "description": "... Backend",
  "main": "built/index.js",
  "scripts": {
    "start": "node_modules/.bin/tsc",
    "watch": "node_modules/.bin/tsc --watch",
    "setupProject": "node_modules/.bin/functions-emulator config set projectId ...",
    "create": "node_modules/.bin/functions-emulator deploy ... --trigger-http",
    "search": "node_modules/.bin/functions-emulator deploy ... --trigger-http",
    "start-emulator": "yarn start && node_modules/.bin/functions-emulator start",
    "stop-emulator": "node_modules/.bin/functions-emulator stop && node_modules/.bin/functions-emulator stop"
  },
  "keywords": [
    "...",
    "..."
  ],
  "author": "rahulrav",
  "license": "MIT",
  "dependencies": {
    "@types/node": "^8.0.31",
    "@types/node-uuid": "^0.0.28",
    "@types/request": "^2.0.3",
    "firebase-admin": "^5.3.0",
    "firebase-functions": "^0.6.3",
    "node-uuid": "^1.4.8",
    "request": "^2.83.0",
    "typescript": "^2.5.3"
  },
  "devDependencies": {
    "@google-cloud/functions-emulator": "^1.0.0-alpha.25",
    "@types/google-cloud__storage": "^1.1.5"
  }
}
```

This is what it looks like. I redacted a few uninteresting parts.

I had to add `@types/google-cloud__storage` to my list of `devDependencies` for the compiler to be happy.

**Comment by hiranya911** [comment]
This should be fixed v5.4.0 and up.
